src/docs/markdown/caddyfile/common-uses.md

```diff
@@ -0,0 +1,122 @@
+---
+title: Common Caddyfile Uses
+---
+
+# Common Caddyfile Uses
+
+This page presents some complete, minimal Caddyfile configurations for common use cases, which you might find helpful when building your own configs.
+
+These are not drop-in solutions; you will have to customize your domain name, ports/sockets, directory paths, etc. They are intended to illustrate only the basic requirements for the most common use cases.
+
+#### Menu
+
+- [Static file server](#static-file-server)
+- [Reverse proxy](#reverse-proxy)
+- [PHP](#php)
+- [Redirect `www.` subdomain](#redirect-www-subdomain)
+- [Trailing slashes](#trailing-slashes)
+
+
+## Static file server
+
+```
+example.com
+
+root * /var/www  # optional; default root is current directory
+file_server
+```
+
+
+## Reverse proxy
+
+All requests:
+
+```
+example.com
+
+reverse_proxy localhost:5000
+```
+
+Just requests having a path starting with `/api/`; static files for everything else:
+
+```
+example.com
+
+reverse_proxy /api/* localhost:5000
+root * /var/www  # optional; default root is current directory
+file_server
+```
+
+
+## PHP
+
+With FastCGI running, something like this works for most modern PHP apps:
+
+```
+example.com
+
+php_fastcgi unix//run/php/php-fpm.sock
+```
+
+If your PHP site relies on static files too, you may need to enable a static file server (but this depends on your PHP app):
+
+```
+example.com
+
+php_fastcgi /blog/* localhost:9000
+root * /var/www  # optional; default root is current directory
+file_server
+```
+
+## Redirect `www.` subdomain
+
+To **add** the `www.` subdomain with an HTTP redirect:
+
+```
+example.com {
+	redir https://www.example.com{uri}
+}
+
+www.example.com {
+	...
```

Comment:
A `root` directive here might be better here, just to make sure no one gets mixed up thinking that `...` is some magic directive.  Same with a few lines later.

Reply:
A root directive by itself is not particularly useful though.

I think -- hope (for the sake of humanity) -- that most people in most cultures know that `...` indicates a continuation or filler of some sort.

Maybe the block should just be left empty, if it's really going to be a problem?

Reply:
I don't feel that strongly either way, but I think `...` is fine. They'll get a syntax error immediately if they try that anyways.

Reply:
How about a `root` and `file_server`, just something minimal that is real?

Reply:
I think that starts distracting from what the example is intending to showcase: the redirect and _only_ the redirect.

I'll leave the block empty then.